pulp_rpm/app/tasks/publishing.py

```diff
@@ -187,24 +212,26 @@ def publish(repository_version_pk, metadata_signing_service=None):
             publication_data.populate()
 
             content = publication.repository_version.content
+            checksums["original"] = repository.original_checksums
 
             # Main repo
             create_repomd_xml(
-                content, publication, publication_data.repomdrecords,
+                content, publication, checksums, publication_data.repomdrecords,
                 metadata_signing_service=metadata_signing_service
             )
 
             for sub_repo in publication_data.sub_repos:
                 name = sub_repo[0]
+                checksums["original"] = sub_repo[-1]
```

Comment:
It's getting harder to read when we use a lot of indexes.
If sub_repo is more complicated than a string, or a dedicated specific object, maybe you can introduce some simple structure for that. E.g. named tuple.